app/models/submission.rb

```diff
@@ -145,12 +145,13 @@ def set_autotest_marks
       if all_marks_earned == 0 || all_marks_total == 0
         final_mark = 0.0
       elsif all_marks_earned > all_marks_total
-        final_mark = mark.markable.max_mark
+        final_mark = mark.criterion.max_mark
       else
-        final_mark = (all_marks_earned / all_marks_total * mark.markable.max_mark).round(2)
-        if mark.markable.instance_of? RubricCriterion
+        final_mark = (all_marks_earned / all_marks_total * mark.criterion.max_mark).round(2)
+        if mark.criterion.instance_of? RubricCriterion
           # find the nearest mark associated to a level
-          nearest_mark = (final_mark / mark.markable.weight.to_f).round * mark.markable.weight
+          # this logic is incorrect due to having an arbitrary number of levels in a rubric criterion
```

Comment:
Nice catch. Mark this as a TODO comment, and also create a new issue to mark this problem. (The problem predates this PR.)